react/LocaleSwitcher.tsx

```diff
@@ -1,54 +1,39 @@
-import React, { useState, FC } from 'react'
-import { useQuery } from 'react-apollo'
-import { useRuntime } from 'vtex.render-runtime'
+import React, { useState } from 'react'
 import { IconGlobe } from 'vtex.store-icons'
 import { useCssHandles } from 'vtex.css-handles'
+import { useRuntime, Culture } from 'vtex.render-runtime'
+import { Spinner } from 'vtex.styleguide'
 
-import Locales from './graphql/locales.gql'
+import LocaleSwitcherList from './components/LocaleSwitcherList'
 
 const CSS_HANDLES = [
-  'container',
-  'relativeContainer',
+  'list',
   'button',
+  'container',
   'buttonText',
-  'list',
   'listElement',
   'localeIdText',
-]
-
-interface LocalesQuery {
-  languages: {
-    default: string
-    supported: string[]
-  }
-  currentBinding: {
-    supportedLocales: string[]
-  } | null
-}
+  'loadingContainer',
+  'relativeContainer',
+] as const
 
-interface SupportedLanguage {
+export interface SupportedLanguage {
   text: string
   localeId: string
 }
 
-function getLabel(localeId: string) {
+export function getLabel(localeId: string) {
```

Comment:
What do you think of moving these functions to a separate module to avoid circular dependency?

It works because of named exports, etc, but it's weird, what do you think?